### PR TITLE
Upload logs and the compose status

### PR DESCRIPTION
**Pull request by gicmo**
Use the new composer API to fetch and upload logs. Additionally also upload the compose status and update it throughout the build.

### CODE REVIEW

File: plugins/builder/osbuild.py [hunk 1]
```diff
@@ -318,8 +318,7 @@ def handler(self, name, version, distro, image_types, target, arches, opts):
 
         # Architectures
         tag_arches = self.arches_for_config(buildconfig)
-        arches = set(arches)
-        diff = arches - tag_arches
+        diff = set(arches) - tag_arches
```
Comment (larskarlitski) [inline plugins/builder/osbuild.py:375]: The commit message says to *not* transform arches to a set, but this code is still doing that.

File: plugins/builder/osbuild.py [hunk 2]
```diff
@@ -203,6 +213,18 @@ def compose_status(self, compose_id: str):
...
+        url = urllib.parse.urljoin(self.url, f"compose/{compose_id}/logs")
+
+        res = self.http.get(url)
+
+        if res.status_code != 200:
+            body = res.content.decode("utf-8").strip()
+            msg = f"Failed to get the compose logs: {body}"
+            raise koji.GenericError(msg) from None
+
+        return ComposeLogs.from_dict(res.json())
```
Comment (larskarlitski) [inline plugins/builder/osbuild.py:237]: Out of curiosity: Is a classmethod conventional in situations like this? I'd have written something like
```python
ComposeLogs(**res.json())
```

**Review by larskarlitski — APPROVED** [review]
Love it! ❤️

Won't "request changes" because of the `set`, but might be something to follow up on.

**Review by teg — COMMENTED** [review]
Very cool!

File: plugins/builder/osbuild.py (hunk 1 above)
Comment (gicmo) [reply plugins/builder/osbuild.py:375]: It does not transform the `arches` variable. Was obviously not clear enough, sorry.

File: plugins/builder/osbuild.py (hunk 2 above)
Comment (gicmo) [reply plugins/builder/osbuild.py:237]: Good question, I *felt* that was a bit more explicit and supports the more nested case better, but I think it is not clear cut.

**Pull request merged**